src/test/java/com/google/api/codegen/testdata/go/go_library.baseline

```diff
@@ -100,11 +100,10 @@ import (
     "google.golang.org/grpc/metadata"
 )
 
-func insertXGoog(ctx context.Context, val []string) context.Context {
-    md, _ := metadata.FromOutgoingContext(ctx)
-    md = md.Copy()
-    md["x-goog-api-client"] = val
-    return metadata.NewOutgoingContext(ctx, md)
+func insertMetadata(ctx context.Context, mds ...metadata.MD) context.Context {
+    out, _ := metadata.FromOutgoingContext(ctx)
+    out = metadata.Join(append([]metadata.MD{out}, mds...)...)
```

Comment:
Was a bit nicer in the library baseline if this accepted an arbitrary number of MDs, but it's a bit ugly here as a result. LMK if you're ok with it or want me to revert.

Reply:
I'm doing this mostly to avoid reconstructing the metadata maps over and over again, see the source of [`metadata.Join`](https://github.com/grpc/grpc-go/blob/master/metadata/metadata.go#L100).

Reply:
You mean for `insertMetadata` to be variadic? I totally agree with this.

`Join` is a little inefficient (I wrote it a while back and still feel a little bad). It's commonly used by multiple goroutines though, so modifying an existing one is a little confusing.